src/Language/utils.php

```diff
@@ -38,14 +40,14 @@ function charCodeAt(string $string, int $position): int
  */
 function printCharCode(int $code): string
 {
-    if ($code === null) {
+    if ($code === 0x0000) {
```

Comment:
This was actually a bug. It appears that `0` equals `<EOF>` in PHP.

Reply:
There is still an issue with this, but I created a separate issue for this (#193).